README.md

```diff
@@ -60,3 +62,44 @@ Then use the `analyticsClient` variable:
+
+Additionally, identify your component using the parameter `componentName`. It's best to keep the value short yet descriptive.
+
+Please ask Operation folks for `#access` to the relevant project in your final destination so you can validate that the tracked events show as expected.
+
+Finally, there is the optional configuration parameter `componentVersion` to track the version of a component sending each event. This is useful to track changes across new component versions so we strongly recommend tracking it.
+
+## Web tracker
+
+[`WebTracker`](src/web.ts) exposes an events tracking interface suitable for web apps. It provides a simple `Page View` event tracking with page loading metrics via `trackPageView`.
+
+When initializing `createWebTracker`, set `prefix` as a short string that describes your project. All events tracked by `WebTracker` will have this `prefix` in their name.
+We use this prefixing method to identify and group events in the final destination for organization purposes and helping the analysts traverse the events with ease.
+This is especially useful when there are multiple components sending events to the same project in the final destination. To give an example, our marketing website, docs pages and blog all send `Page View` event but these are prefixed respectively `Marketing`, `Docs` and `Blog`. So the final event name is `[Marketing] Page View`, `[Docs] Page View`, etc. Furthermore, this method makes it easier to identify all the `Marketing` events.
+
+## Identifying users
+
+By default the client tracks `deviceId` to identify the current device of the end user. `deviceId` is a UUID generated automatically by the client. Use `setUserId` method to identify authenticated user. At least one of these identifiers is required to successfully track an event.
+
+In most cases the device refers to the browser of the user (or Electron application for the case of Etcher). By persisting the `deviceId` in the local storage, we are able to identify the events coming from the same user in order to create a timeline of component usage.
+
+It is possible to generate a new device identifier and also override an existing device identifier. We use these methods especially when we need to link multiple device identifiers that we know belonging to the same user.
+For instance, a balenaCloud user might login to the dashboard from multiple browsers and also use balenaCLI. We would like to create a coherent timeline that captures all their activity and display it under the same user in order to make a sound analysis of their balena platform usage. As long as all the components `setUserId` as soon as the authentication happens, we are able to link all the events with separate `deviceId` values that came *even before* the authentication on the backend side.
```

Comment:
How does it synergize with the merging abilities of Amplitude? Or are we reimplementing the same thing?

Reply:
We are utilizing Amplitude SDK's methods to achieve these actually. The main logic for merging sits in url-params.ts file. There we parse the URL parameters to link multiple device ids to the same user id.